### ISSUE TITLE
create plugin to infer (biome) task based on config presence

### ISSUE THREAD

**Issue by getlarge** [GitOpsLovers/nx-biome#1]
Hey, I am currently interested in using Biome in some Nx workspaces and discovered your plugin. Thanks for starting this work.

What would think of adding the [Project Crystal sauce](https://nx.dev/concepts/inferred-tasks) in your plugin so that tasks are "automatically" added to Nx projects declaring a Biome config?

I would be willing to contribute.

**Comment by mcfdez** [comment]
Hello @getlarge 

I have not been able to study this Project Crystal feature in depth yet, but it sounds interesting, so I am open to contributions.

Do you have any idea in mind how the plugin could operate?

**Comment by getlarge** [comment]
Yes, it would follow the typical flow already implemented by several plugins such as ESLint or Jest.

From the users' perspective:
 1. Declare the plugin in `nx.json`, with a `targetName` option (maybe `biome-lint` as default to avoid clashing with the usual `lint` target for ESLint).
 2. Check the target was added by running `nx show projects <projectName>`
 
 From the developers' perspective:
 1. Check the `createNodes` API usage [here](https://nx.dev/extending-nx/recipes/project-graph-plugins#example-extending-projects-adding-targets) to add dynamic targets.
 2. add logic to check for biome config files (can be globs) presence in a project
 3. create a dynamic target based on the `targetName` option
 
 Does it answer your question?

**Comment by nacho-vazquez** [comment]
I'm open to implement this feature.
